0.0.1/agent.py

```diff
@@ -1,93 +1,188 @@
+        tool_registry.register_tool(self.save_near_account_id)
+        tool_registry.register_tool(self.get_goals)
+        tool_registry.register_tool(self.save_goal)
+        tool_registry.register_tool(self.get_near_account_balance)
+        tool_registry.register_tool(self.fetch_token_prices)
+
+    @property
+    def near_account_id(self) -> str | None:
+        if not self._near_account_id:
+            self._near_account_id = self.env.read_file("near_id.txt")
+        return self._near_account_id
+
+    @property
+    def allowance_goal(self) -> str | None:
+        if not self._allowance_goal:
+            self._allowance_goal = self.env.read_file("allowance_goal.txt")
+        return self._allowance_goal
+
+    @property
+    def growth_goal(self) -> str | None:
+        if not self._growth_goal:
+            self._growth_goal = self.env.read_file("growth_goal.txt")
+        return self._growth_goal
 
     def run(self):
         # A system message guides an agent to solve specific tasks.
-        prompt = {"role": "system", "content": "You are an assistant that helps people set goals for growth in the USD value of their crypto assets such that when that percentage in growth has been reached or surpassed, you look at their tokens and determine the tokens and quantities of each to swap for USDT stablecoins or USDC stablecoins"}
+        prompt = {
+            "role": "system",
+            # disallow some builtin tools from the library
+            "content": """
+You are Divvy, a financial assistant that helps users manage and grow their crypto portfolio.
+You can access NEAR account details of the user (their balance and account id).
+You can provide the real-time current market prices of crypto tokens in the users wallet.
+You can allow the user to set growth and allowance goals on their portfolio.
+
+Your capabilities are defined and facilitated by the tools you have access to except your disallowed tools.
+
+-Disallowed tools-
+`list_files, query_vector_store, write_file, request_user_input, read_file, exec_command`.
+
+You must follow the following instructions:
+
+-Instructions-
+* Be polite and helpful to the user.
+* When introducing yourself, provide a brief description of what your purpose is.
+* Tell the user if you don't support a capability. Do NOT make up or provide false information or figures.
+* Do not expose the functions you have access to to the user.
+* Do not use figures or function call from preceding messages to generate responses.
+""",
+        }
 
         # Use the model set in the metadata to generate a response
-        result = self.env.completion([prompt] + self.env.list_messages())
+        result = None
+        tools = self.env.get_tool_registry().get_all_tool_definitions()
+        tools_completion = self.env.completion_and_get_tools_calls(
+            [prompt] + [self.env.get_last_message() or ""], tools=tools
+        )
+        self.env.add_system_log(
```

Comment:
@RyanSept I'm not seeing any of these debug logs. I've tried passing an env var: --env_vars='{"DEBUG":"true"}'.  Might you have another suggestion?

Reply:
This lib is a mess. These logs show up in this tmp file not in stdout!
https://github.com/beneviolabs/ft-allowance-agent/pull/11/files#diff-0242332b677ca693d33c29815b6dea020533ebc1d56b7097801b77b3ef8db819R54

Reply:
aaaahhhh !